src/utilities/strings.ts

```diff
@@ -112,3 +112,80 @@ export const camelCase = (string: string): string => {
       return index === 0 ? match.toLowerCase() : match.toUpperCase()
     })
 }
+
+// Taken from the React escapeTextForBrowser internal utility
+const escapeHtmlRegExp = /["'&<>]/
+
+/**
+ * Escape HTML special characters in the string for output in the browser.
+ *
+ * @param {string} string
+ * @returns {string}
+ */
+export const escapeHTML = (string: string): string => {
+  const match = escapeHtmlRegExp.exec(string)
+
+  if (!match) {
+    return string
+  }
+
+  let escape
+  let html = ''
+  let index
+  let lastIndex = 0
+
+  for (index = match.index; index < string.length; index++) {
+    switch (string.charCodeAt(index)) {
+      case 34: // "
+        escape = '&quot;'
+        break
+      case 38: // &
+        escape = '&amp;'
+        break
+      case 39: // '
+        escape = '&#x27;'
+        break
+      case 60: // <
+        escape = '&lt;'
+        break
+      case 62: // >
+        escape = '&gt;'
+        break
+      default:
+        continue
+    }
+
+    if (lastIndex !== index) {
+      html += string.substring(lastIndex, index)
+    }
+
+    lastIndex = index + 1
+    html += escape
+  }
+
+  return lastIndex !== index ? html + string.substring(lastIndex, index) : html
+}
+
+/**
+ * Autolinks URLs in the given string.
+ *
+ * @param {string} string
+ *
+ * @returns {string}
+ */
+export const autolink = (string: string): string => {
```

Comment:
@tompedals Looking at the name in isolation, I'm not sure what `autolink` does. I mean, I know what it does after :pear:'ing, but it's not as clear as something like `escapeHTML`.

Would you be okay with renaming `autolink` to something like `convertLinksToHTML`?

It's a little bit more verbose, but it's a lot clearer (to me at least!~)